vcbuild.bat

```diff
@@ -1,5 +1,7 @@
 @if not defined DEBUG_HELPER @ECHO OFF
 
+setlocal
```

Comment:
So, this is a "semver-major" change. Several build related scripts depend on several variables ability to escape.

It might be a simpler approach to just add an escape hatch %extra_msbuild_args% arg to L316, and keep this as a documentation only feature.

Reply:
This is what this review is for. Thanks for catching that! :)

I don't actually recall the exact error I hit that prompted this, but it was of the form 'input arg causes env var to be set, env var isn't cleared, arg can't be unset'. If I can figure out what that was, then from your description I would be wiser to add it to the env var reset block at the top of the file. That way the environment would still be modified, but re-running the script wouldn't hit...whatever problem I was hitting.

Would it make sense to replace this line with

    @rem Some other scripts rely on the environment variables set 
    @rem in this script. Allow this to happen.
    @rem setlocal

Reply:
Were you suggesting `%extra_msbuild_args%` as a different way to implement the binlogging part, or in connection to setlocal?

(The more I stare at this the more it feels like it should be two PRs...)

Reply:
> I don't actually recall the exact error I hit that prompted this

I'd assume the the state of `DEBUG_HELPER` will become persistent...

> Were you suggesting `%extra_msbuild_args%` as a different way to implement the binlogging part, or in connection to setlocal?

Yeah, as an alternative of adding a `verbose` flag (and all it's side effects).

Reply:
> Would it make sense to replace this line with

Good idea.
I'd suggest:
```cmd
:: Other scripts rely on the environment variables set in this script. So we explicitly allow them to escape.
endlocal
```

Reply:
Ha! You're right! The 'verbose' arg was setting DEBUG_HELPER, and DEBUG_HELPER leaking into non-'verbose' invocations. 

I can think of two tidy options:
1. Add a 'binlog' arg that sets its own variable and just adds the binlog arguments to the msbuild call
2. Add an extra_msbuild_arg arg that takes the next token and passes it as-is to msbuild

I prefer the first because I don't have to remember the binary log arguments for msbuild, but I'd be happy with the second.

Reply:
Let me know what you think of the recent revisions.